tool/internal/instrument/trampoline.go

```diff
@@ -676,37 +687,81 @@ func (ip *InstrumentPhase) rewriteHookContext() {
-			clause := setParamClause(idx, paramType)
-			methodSetParamBody.List = append(methodSetParamBody.List, clause)
-			clause = getParamClause(idx, paramType)
-			methodGetParamBody.List = append(methodGetParamBody.List, clause)
-			idx++
-		}
+	splitParams := ast.SplitMultiNameFields(ip.targetFunc.Type.Params)
+	for _, param := range splitParams.List {
+		paramType := replaceTypeParamsWithAny(desugarType(param), ip.targetFunc.Type.TypeParams)
+		clause := setParamClause(idx, paramType)
+		methodSetParamBody.List = append(methodSetParamBody.List, clause)
+		clause = getParamClause(idx, paramType)
+		methodGetParamBody.List = append(methodGetParamBody.List, clause)
+		idx++
 	}
 	// Rewrite GetReturnVal and SetReturnVal methods
 	if ip.targetFunc.Type.Results != nil {
 		idx = 0
-		for _, retval := range ip.targetFunc.Type.Results.List {
-			retType := desugarType(retval)
-			for range retval.Names {
-				clause := getReturnValClause(idx, retType)
-				methodGetRetValBody.List = append(methodGetRetValBody.List, clause)
-				clause = setReturnValClause(idx, retType)
-				methodSetRetValBody.List = append(methodSetRetValBody.List, clause)
-				idx++
+		splitResults := ast.SplitMultiNameFields(ip.targetFunc.Type.Results)
+		for _, retval := range splitResults.List {
+			retType := replaceTypeParamsWithAny(desugarType(retval), ip.targetFunc.Type.TypeParams)
+			clause := getReturnValClause(idx, retType)
+			methodGetRetValBody.List = append(methodGetRetValBody.List, clause)
+			clause = setReturnValClause(idx, retType)
+			methodSetRetValBody.List = append(methodSetRetValBody.List, clause)
+			idx++
+		}
+	}
+}
+
+// isTypeParameter checks if a type expression is a bare type parameter identifier
+func isTypeParameter(t dst.Expr, typeParams *dst.FieldList) bool {
+	if typeParams == nil {
+		return false
+	}
+	ident, ok := t.(*dst.Ident)
+	if !ok {
+		return false
+	}
+	// Check if this identifier matches any type parameter name
+	for _, field := range typeParams.List {
+		for _, name := range field.Names {
+			if name.Name == ident.Name {
+				return true
 			}
 		}
 	}
+	return false
+}
+
+// replaceTypeParamsWithAny replaces type parameters with interface{} for use in
+// non-generic contexts like HookContextImpl methods
+func replaceTypeParamsWithAny(t dst.Expr, typeParams *dst.FieldList) dst.Expr {
+	if isTypeParameter(t, typeParams) {
+		return ast.InterfaceType()
+	}
+
+	// For complex types like *T, []T, map[K]V, etc., handle them recursively
+	switch tType := t.(type) {
+	case *dst.StarExpr:
+		// *T -> *interface{}
+		return ast.DereferenceOf(replaceTypeParamsWithAny(tType.X, typeParams))
+	case *dst.ArrayType:
+		// []T -> []interface{}
+		return ast.ArrayType(replaceTypeParamsWithAny(tType.Elt, typeParams))
+	case *dst.MapType:
+		// map[K]V -> map[interface{}]interface{}
+		return &dst.MapType{
+			Key:   replaceTypeParamsWithAny(tType.Key, typeParams),
+			Value: replaceTypeParamsWithAny(tType.Value, typeParams),
+		}
+	}
```

Comment:
Does this cover all possible syntax cases? Should we add a default case with util.Unimplemented() or util.ShouldNotReachHere() to be safe?